source/elements/oneDPL/source/common.rst

```diff
@@ -2,14 +2,51 @@
 ..
 .. SPDX-License-Identifier: CC-BY-4.0
 
+General Information
+-------------------
+
+This section describes the most general functionality of |dpl_full_name| (oneDPL)
+such as namespaces, versioning, etc.
+
 Namespaces
-----------
+++++++++++
 
 oneDPL uses ``namespace oneapi::dpl`` and a shorter variant ``namespace dpl`` for all
 functionality including parallel algorithms, oneDPL execution policies, etc.
 For the subset of the standard C++ library for kernels, the standard class
 and function names are also aliased in ``namespace oneapi::dpl``.
 
 oneDPL uses nested namespaces for the functionality aligned with the C++ standard.
-The names of those namespaces are the same as in ``namespace std``. For example,
-oneDPL execution policies are provided in ``namespace oneapi::dpl::execution``.
+The names of those namespaces are the same as for their analogues in ``namespace std``.
+[*Example*: oneDPL execution policies are provided in ``namespace oneapi::dpl::execution``.  -- *end example*]
+
+Header Files
+++++++++++++
+
+The oneDPL header files are provided under the ``oneapi/dpl/`` include path,
+which must explicitly precede oneDPL header file names in ``#include`` directives.
+
+Following the naming convention for the C++ standard library header files,
+the oneDPL header files have no extension suffix in their names.
+
+A oneDPL header file automatically includes the C++ standard library header file
+which name matches that of the oneDPL header file without the include path prefix.
+[*Example*: ``#include <oneapi/dpl/version>`` automatically includes ``<version>`` 
+if that is available. -- *end example*]
+
+Version Information
++++++++++++++++++++
+
+A oneDPL implementation must define a preprocessor macro representing the version
+of the oneDPL specification that the implementation is compliant with.
+
+.. code:: cpp
+
+  // Defined in <oneapi/dpl/version>
+
+  #define ONEDPL_SPEC_VERSION /*implementation-defined*/
+
+The ``ONEDPL_SPEC_VERSION`` macro must be defined to the decimal literal which value equals to
+*major-spec-version* * 100 + *minor-spec-version*, where *major-spec-version* and *minor-spec-version*
+are the major and the minor versions of the latest fully supported specification.
+[*Example*: ``#define ONEDPL_SPEC_VERSION 104`` for the oneDPL specification 1.4.  -- *end example*]
```

Comment:
I don't think there is a oneDPL specification - it is only oneAPI specification, which defines versioning in terms of major, minor and revision, not in the format `1.4`. Can you please clarify?

Reply:
With all that said, if a proposal comes how the wording can be improved, I will happily consider and apply it.

Reply:
@akukanov 
>> By default, elements do not have independent versioning. ... Other elements can be split out and have independent versioning if the need arises with agreement of the oneAPI core team.

> and there were precedents in the past: Level Zero and oneVPL, before being removed from the oneAPI spec, maintained their versions independently.

I would extend the question - if we separate so many chapters, what will remain under the oneAPI umbrella, considering the software architecture: https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.3-rev-1/architecture#software-architecture ?

Reply:
@vmalia is it a "what if" question? Maybe I do not get it, but so far nothing is separated, is it?

Reply:
Currently we have the same version numbering and scheme used across the entire oneAPI spec, except Level Zero like you mentioned. As I see you are suggesting splitting oneDPL's versioning scheme from oneAPI-spec, is that right?

My question is - what is stopping other chapters from doing the same thing? Is there a specific reason/method we use to decide if a chapter is eligible for "independent versioning"?

Reply:
> As I see you are suggesting splitting oneDPL's versioning scheme from oneAPI-spec, is that right?

No, I am not. May I ask which exactly part of the wording led you to such a conclusion?
I just find it fully appropriate to refer to the oneDPL specification and its version, whether it is aligned with the version of the whole oneAPI specification (as now) or not. More generally, I view the oneAPI specification as modular, not monolithic. Each element chapter is the specification for that element/component, while the oneAPI specification is a collection of those.

> what is stopping other chapters from doing the same thing? Is there a specific reason/method we use to decide if a chapter is eligible for "independent versioning"?

As I pointed out, I interpret the versioning rules as allowing, in principle, independent versioning of the specifications for oneAPI elements. The method, I believe, will be just as for any other specification change - someone need to make an RFC proposal stating the reasons as well as some analysis, and get the RFC approved. And I interpret "with agreement of the oneAPI core team" in the quote above as "must be approved by the core team".

Reply:
Added the following wording to the beginning of the oneDPL chapter:

> This is the specification of oneAPI DPC++ Library, part of the oneAPI Specification version 1.4-provisional-rev-1.

Hopefully this clarifies the intent.

Reply:
>> As I see you are suggesting splitting oneDPL's versioning scheme from oneAPI-spec, is that right?

> No, I am not. May I ask which exactly part of the wording led you to such a conclusion?
```rst
[*Example*: ``#define ONEDPL_SPEC_VERSION 104`` for the oneDPL specification 1.4.  -- *end example*]
```
It was not obvious to me that oneAPI and oneDPL (or any other chapter names) can be used interchangeably when it comes to versioning. It should be clarified somewhere. That being said, I am okay with the idea overall.